### ISSUE TITLE
Deprecate flush_size and use pipeline batch size as ES bulk request size

### ISSUE THREAD

**Issue by suyograo** [logstash-plugins/logstash-output-elasticsearch#382]
There are 2 different knobs to tune batching in Logstash:
* The pipeline batch size which was introduced in 2.2 that controls the overall event batching for *all* filters and outputs.
* Additionally, in ES output we have `flush_size` which controls the bulk request size to Elasticsearch.

Proposal is to deprecate `flush_size` and just use pipeline batch size. It is just confusing to have to tune batching in multiple places. This also plays well with future acking proposal for resiliency. There has been asks for flush size based on bytes, but IMO, this size based buffering should be also at the pipeline level.

A big drawback to this approach is the one-size-fits-all for buffering events in outputs, which may not work well for certain outputs like HDFS, S3, etc. 

Moved this discussion from https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/381#discussion_r54793561

Thoughts?

**Comment by jsvd** [comment]
Relevant discussion on the ES side: https://github.com/elastic/elasticsearch/issues/16011

**Comment by jordansissel** [comment]
Alternate solution is provided by  https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/497

**Comment by dlgasser** [comment]
#497 doesn't seem to be a fix for users of AWS ElasticSearch, which caps bulk requests at 10Mb.

**Comment by jordansissel** [comment]
@dlgasser Does this plugin even work with Amazon Elasticsearch Service? I think Amazon forked this repository a long time ago and are maintaining their own plugin specific to Amazon's Elasticsearch service.

**Comment by dlgasser** [comment]
Yes, it still works if your IAM setup doesn't require connection requests to be signed (i.e. you use network or EC2 role-based access control for the ElasticSearch service)

**Comment by jordansissel** [comment]
I am hesitant to make this tunable. However, @dlgasser can you open a separate issue to discuss your specific issue? (10mb bulk cap)

**Comment by RishikeshDarandale** [comment]
@jordansissel 

> I am hesitant to make this tunable. However, @dlgasser can you open a separate issue to discuss your specific issue? (10mb bulk cap)

I see that [#785](https://github.com/logstash-plugins/logstash-output-elasticsearch/issues/785) issue is open against above.
